### DISCUSSION TITLE
Generic maths

### DISCUSSION THREAD

**Discussion by Xyncgas** [fsharp/fslang-suggestions#1173] [General]
Say there is a function that takes two number, in the function there is :
~ Operation that uses a constant programmer wrote in their favorite editor, for example ``` 1 ```
~ Comparison, Addition, Multiplication, Division etc with the parameter themselves ``` if param1 < param2 then ... ```
~ Mathematical operations between the parameter and the constant, for example ``` param1 <- param1 + 1 ```

It would be nice that param1 doesn't get inferred to int when we are doing ``` param1 <- param1 + 1 ``` because 1 is not a negative number and we can do this with ``` float ``` ``` int ``` ``` uint ``` ``` int64```...

Edit :I've found helped in the ```LanguagePrimitives``` namespace

**Comment by vzarytovskii** [comment]
Hey, thanks for the question, however it's not a proper place for such questions, since not that many people are following this repo.

You'll have better luck asking it on SO, slack or even official repository (dotnet/fsharp).

If you, however, would like to make a suggestion for the language or the standard library, please use the official issue template instead.

Thanks.
